src/main/kotlin/com/viartemev/thewhiterabbit/consumer/ConfirmConsumer.kt

```diff
@@ -36,12 +33,22 @@ class ConfirmConsumer internal constructor(private val AMQPChannel: Channel, AMQ
         )
     }
 
-    suspend fun consumeWithConfirm(handler: suspend (Delivery) -> Unit, handlerDispatcher: CoroutineDispatcher = Dispatchers.Default) = coroutineScope {
+    suspend fun consumeWithConfirm(handler: suspend (Delivery) -> Unit, handlerDispatcher: CoroutineDispatcher = Dispatchers.Default) {
         val delivery = continuations.receive()
+        val deliveryTag = delivery.envelope.deliveryTag
         withContext(handlerDispatcher) { handler(delivery) }
-        AMQPChannel.basicAck(delivery.envelope.deliveryTag, false)
+        try {
+            AMQPChannel.basicAck(deliveryTag, false)
+        } catch (e: Exception) {
+            val errorMessage = "Can't ack a message with deliveryTag: $deliveryTag"
```

Comment:
Is it truth that if we've got exception then it's ack? What about ACK + IO exception?

Reply:
We use try-catch only for basicAck. A user is responsible for handling exceptions which can be thrown from the handler.